### PR TITLE
Dynamic sanctions

### PR DESCRIPTION
**Pull request by antonypap**

### CODE REVIEW

**Comment by antonypap** [comment]
`activeSanctionMap` sanctions are always false, although agents are being pruned...? any ideas? (`sanctionHistory` seems to work fine)

File: pkg/infra/teams/team3/leader_strat.go [hunk 1]
```diff
@@ -150,13 +177,62 @@ func (a *AgentThree) Sanctioning(agent agent.Agent) int {
 	return sanction
 }
 
+func (a *AgentThree) sanctioningHistoric(agent agent.Agent) int {
+	agentId := agent.ID()
+	prevSanctions := a.sanctionHistory[agentId]
+	mostRecentSanction := prevSanctions[len(prevSanctions)-1]
+	return mostRecentSanction + 1
+}
```
Comment (antonypap) [inline pkg/infra/teams/team3/leader_strat.go:191]: is that supposed to be used to calculate dynamic sanctioning? (in `sanctioningDynamic()`)

**Comment by MattSScott** [comment]
> `activeSanctionMap` sanctions are always false, although agents are being pruned...? any ideas? (`sanctionHistory` seems to work fine)

Weird. I'll run some tests tomorrow morning

**Comment by antonypap** [comment]
Should be good for now

**Pull request merged**